src/SignalR/common/Protocols.Json/src/Protocol/JsonHubProtocol.cs

```diff
@@ -187,7 +188,25 @@ public ReadOnlyMemory<byte> GetMessageBytes(HubMessage message)
                         }
                         else if (reader.ValueTextEquals(TargetPropertyNameBytes.EncodedUtf8Bytes))
                         {
+#if NETCOREAPP
+                            reader.Read();
+
+                            if (reader.TokenType != JsonTokenType.String)
+                            {
+                                throw new InvalidDataException($"Expected '{TargetPropertyName}' to be of type {JsonTokenType.String}.");
+                            }
+
+                            if (!reader.HasValueSequence)
+                            {
+                                target = binder.GetTarget(reader.ValueSpan) ?? Encoding.UTF8.GetString(reader.ValueSpan);
```

Comment:
Are we at all worried about `\` escaping in the JSON target string? Is it possible that there are any clients that leverage this for certain unicode characters?

https://github.com/dotnet/runtime/blob/96db556687a08b65982abbfc76256e7312ffe994/src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs#L41-L46

It doesn't seem super likely since the JSON spec requires UTF8 encoding/decoding support anyway, but it seems odd not to support `\` escaping when the target happens to arrive in a single segment of the `ReadOnlySequence<byte>` but then support it in the when the target arrives in multiple segments.

Reply:
I'm skipping it because it's extra work, there's no non-allocating API and it'll fallback to doing that because it won't find the string anyways.

Reply:
Ah wait, you're right that I need to fix the fallback here. I'll look into that part.